### PR TITLE
add authorisation to GET/api/profiles

### PR DESCRIPTION
**Pull request by mara-ber**

### CODE REVIEW

File: server/api/profiles.js [hunk 1]
```diff
@@ -38,16 +38,24 @@ router.post(
...
+      getAllProfiles()
+        .then(data => {
+          res.send(data);
+        })
+        .catch(err => {
+          console.error(err);
+          res.send(500);
+        });
+    } else {
+      res.send("you falled authorisation");
```
Comment (gregdyke) [inline server/api/profiles.js]: This should send a 403 Forbidden Status

**Review by gregdyke — COMMENTED** [review]
You also need to change /client/api/profiles.js so that is adds in the authorisation header for this request

File: server/api/profiles.js [hunk 2]
```diff
@@ -38,16 +38,24 @@ router.post(
...
+  passport.authenticate("jwt", { session: false }),
+  (req, res) => {
+    if (req.user.role === "recruiter") {
+      getAllProfiles()
+        .then(data => {
+          res.send(data);
+        })
+        .catch(err => {
+          console.error(err);
+          res.send(500);
```
Comment (alastair87) [inline server/api/profiles.js:52]: This is probably not what you want as it doesn't send a response status of 500, it send `500` as the response. I suggest changing it to either `res.sendStatus(500)` or `res.status(500).send('some description here')`.

File: server/api/profiles.js [hunk 3]
```diff
@@ -38,16 +38,24 @@ router.post(
...
+        });
+    } else {
+      res.sendStatus(403);
+    }
```
Comment (alastair87) [inline server/api/profiles.js:56]: You might want to include more information like before e.g. `res.status(403).send("you failed authorisation")` however this is not essential.

File: server/api/profiles.js (hunk 2 above)
Comment (gregdyke) [reply server/api/profiles.js:52]: res.send(Number) actually will res.sendStatus, whereas res.send(String), which is why we have some res.send(500) sitting around in the codebase and probably the syllabus. These should more cleanly be res.sendStatus(500) :)

File: server/api/profiles.js (hunk 3 above)
Comment (gregdyke) [reply server/api/profiles.js:56]: If we want to add a more useful message, maybe "user in token in authorisation header must have the recruiter role" (403 already tells us we failed authorisation)

**Review by gregdyke — APPROVED** [review]

**Pull request merged**